src/scheduler/FunctionCallClient.cpp

```diff
@@ -123,29 +167,53 @@ void FunctionCallClient::sendMPIMessage(
     if (faabric::util::isMockMode()) {
         mpiMessages.emplace_back(host, *msg);
     } else {
-        ClientContext context;
-        faabric::FunctionStatusResponse response;
-        CHECK_RPC("mpi_message", stub->MPICall(&context, *msg, &response));
+        // Send the header first
+        sendHeader(faabric::scheduler::FunctionCalls::MpiMessage);
+
+        // Send the message body
+        size_t msgSize = msg->ByteSizeLong();
+        char* serialisedMsg = new char[msgSize];
+        // Serialise using protobuf
+        if (!msg->SerializeToArray(serialisedMsg, msgSize)) {
+            throw std::runtime_error("Error serialising message");
+        }
+        send(serialisedMsg, msgSize);
     }
 }
 
-faabric::HostResources FunctionCallClient::getResources(
-  const faabric::ResourceRequest& req)
```

Comment:
This request was empty, and we needed not to create it outside the function's scope.